### ISSUE TITLE
`backend` namespace collision

### ISSUE THREAD

**Issue by jo** [eHealthAfrica/kazana-config#27]
In spawn-pouchdb-server the [`backend` option](https://github.com/hoodiehq/spawn-pouchdb-server#options) can be `false`, in which case the in memory adapter is used.

In our case we cannot set `backend` to false because it is extended and also includes `log`, `timeout` and `port` options.

I suggest we replace the `backend` option with `db` and move it to the inside, providing only the options spawn-pouchdb-server understands:

```js
{
  db: {
    backend: {
      name: 'leveldown',
      location: './.db'
    }, // or `false`
    log: {
      file: 'asd',
      level: 'silly'
    },
    config: {
      file: './.db/config.json'
    },
    timeout: 1234
  }
}
```

**Comment by jo** [comment]
For reference, this is now:

```js
{
  backend: {
    name: 'leveldown',
    location: './.db',
    log: {
      file: 'asd',
      level: 'silly'
    },
    config: {
      file: './.db/config.json'
    },
    timeout: 1234
}
```

**Comment by gr2m** [comment]
yes, I like :+1:

**Comment by patriciagarcia** [comment]
@jo, can we close this? (is this solved by the other PR that set up the in memory option?)

**Comment by jo** [comment]
no, this needs to be implemented. It will be a breaking change.

**Comment by gr2m** [comment]
I would prefer if we could check if `config.backend` is set, then adapt it on the fly and show a deprecation warning. I'd try to avoid breaking changes if not needed, ideally a new major version means we remove the deprecation warnings and obsolete APIs. That makes it simpler to keep our apps up-to-date, and the developers can do the migrations themselves when they see the warnings in the logs

**Comment by patriciagarcia** [comment]
Sounds good, maybe we can keep an issue with deprecated APIs/behaviour to be removed in the next major release? This way we won't forget about it and we'll keep the code clean while not breaking kazana too often.

**Comment by patriciagarcia** [comment]
Here is the aforementioned issue https://github.com/eHealthAfrica/kazana/issues/113

**Comment by jo** [comment]
Should be done like this: https://github.com/eHealthAfrica/kazana-config/pull/42

**Comment by patriciagarcia** [comment]
@jo, can this be closed then? (is it solved in #42?)

**Comment by gr2m** [comment]
no it’s not done, #42 was only related. This issue about moving the options that `spawn-pouchdb-server` takes into it’s own namespace, Johnannes suggested `kazana.db.backend` instead of `kazana.backend`, so that we can set it to false. We can’t set the option to false right now, because we need other options on it like port & log level.

With "Should be done like this: #42" @jo suggests that we implement the change as we did with adding the new `kazana` config namespace, by adding deprecation warnings, but without breaking existing apps

**Comment by patriciagarcia** [comment]
Ok, thanks for the explanation Gregor! :)
